internal/command/jobs/printers.go

```diff
@@ -0,0 +1,213 @@
+package jobs
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"text/tabwriter"
+	"time"
+
+	"github.com/signadot/cli/internal/config"
+	"github.com/signadot/cli/internal/sdtab"
+	"github.com/signadot/go-sdk/client/artifacts"
+	"github.com/signadot/go-sdk/models"
+	"github.com/xeonx/timeago"
+)
+
+type jobRow struct {
+	Name        string `sdtab:"NAME"`
+	Environment string `sdtab:"ENVIRONMENT"`
+	StartedAt   string `sdtab:"STARTED AT"`
+	Duration    string `sdtab:"DURATION"`
+	Status      string `sdtab:"STATUS"`
+}
+
+func printJobTable(cfg *config.JobList, out io.Writer, jobs []*models.JobsJob) error {
+	t := sdtab.New[jobRow](out)
+	t.AddHeader()
+
+	sort.Slice(jobs, func(i, j int) bool {
+		if jobs[i].Status.Attempts[0].Phase == "queued" {
+			return true
+		}
+
+		if jobs[j].Status.Attempts[0].Phase == "queued" {
+			return false
+		}
+
+		t1, err1 := time.Parse(time.RFC3339, jobs[i].CreatedAt)
+		t2, err2 := time.Parse(time.RFC3339, jobs[j].CreatedAt)
+		if err1 != nil || err2 != nil {
+			return false
+		}
+
+		return t2.Before(t1)
+	})
+	for _, job := range jobs {
+		startedAt, duration := getStartedAndDuration(job)
+
+		environment := ""
+		routingContext := job.Spec.RoutingContext
+		switch {
+		case routingContext == nil:
+		case len(routingContext.Sandbox) > 0:
+			environment = fmt.Sprintf("sandbox=%s", routingContext.Sandbox)
+		case len(routingContext.Routegroup) > 0:
+			environment += fmt.Sprintf("routegroup=%s", routingContext.Routegroup)
+		}
+
+		t.AddRow(jobRow{
+			Name:        job.Name,
+			Environment: environment,
+			StartedAt:   startedAt,
+			Duration:    duration,
+			Status:      job.Status.Phase,
+		})
+	}
+	return t.Flush()
+}
+
+func printJobDetails(cfg *config.Job, out io.Writer, job *models.JobsJob) error {
+	tw := tabwriter.NewWriter(out, 0, 0, 3, ' ', 0)
+
+	startedAt, duration := getStartedAndDuration(job)
+
+	fmt.Fprintf(tw, "Job Prefix:\t%s\n", job.Spec.NamePrefix)
+	fmt.Fprintf(tw, "Job Name:\t%s\n", job.Name)
+	fmt.Fprintf(tw, "Status:\t%s\n", job.Status.Phase)
```

Comment:
Wonder whether to call it `phase` (which is what it is as per the API) or `status` (keeping it in-line with statuses for other entities). @foxish ? I think the former might be preferable to alleviate confusion.

Reply:
sgtm